src/impl.h

```diff
@@ -215,8 +211,14 @@ struct Manifold::Impl {
     CreateFaces();
 
     SimplifyTopology();
+    RemoveUnreferencedVerts();
```

Comment:
This was the fix for the reported bug: it seems that in some cases `SimplifyTopology` strands unreferenced verts that it does not set to NAN. Ideally it shouldn't do that - probably worth looking into in the future; it needs an overhaul anyway.